polyply/src/generate_templates.py

```diff
@@ -349,19 +352,24 @@ def _gen_templates(self, meta_molecule):
                         break
                     else:
                         opt_counter += 1
-
-                self.volumes[resname] = compute_volume(meta_molecule, block, coords, self.topology.nonbond_params)
                 coords = map_from_CoG(coords)
                 self.templates[resname] = coords
+            elif resname in meta_molecule.templates:
+                coords = meta_molecule.templates[resname]
+                coords = map_from_CoG(coords)
+                self.templates[resname] = coords
+            else:
+                coords = None
 
-        return templates, volumes
```

Comment:
Is it intentional this method doesn't return anything anymore?

Reply:
yes in fact the function updates some class attributes